library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/MaskInputEditTextViewMapper.kt

```diff
@@ -0,0 +1,21 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder.mapper
+
+import com.datadog.android.sessionreplay.internal.recorder.obfuscator.FixedLengthStringObfuscator
+import com.datadog.android.sessionreplay.utils.StringUtils
+import com.datadog.android.sessionreplay.utils.UniqueIdentifierGenerator
+import com.datadog.android.sessionreplay.utils.ViewUtils
+
+internal class MaskInputEditTextViewMapper(
```

Comment:
I have a concern here (maybe it is related to the architecture) - class which does nothing, but calls another class, even though with some custom parameters look a bit strange and non-obvious. For example here, having a values which can be set + default arguments imply that we can for example pass `textViewMapper` which won't mask anything and then class name (`MaskInputEditTextViewMapper`) won't match the actual functionality?

Reply:
yes...the non-default arguments are meant for testing only, this class should only encapsulate that `TextViewMapper` with those `FixedLengthObfuscators`. Would it be more visible if instead of using a generic `TextViewMapper` I create one `MaskInputTextViewMapper` which will be used by the `MaskInputEditTextViewMapper` ? 

Have to admit that I am wondering myself what kind of architecture I should use in this situation so any suggestion would help with the small note that I don't want to change this too much as the rules are not yet 100% detailed/approved and these might change later.

Reply:
I actually decided to try to change it from the ground. I am going to come up with a new commit on top of this one with a proposal later. Will keep you posted.